google/cloud/spanner/admin/integration_tests/backup_extra_integration_test.cc

```diff
@@ -0,0 +1,483 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "google/cloud/spanner/admin/database_admin_client.h"
+#include "google/cloud/spanner/admin/database_admin_options.h"
+#include "google/cloud/spanner/admin/instance_admin_client.h"
+#include "google/cloud/spanner/backoff_policy.h"
+#include "google/cloud/spanner/backup.h"
+#include "google/cloud/spanner/client.h"
+#include "google/cloud/spanner/polling_policy.h"
+#include "google/cloud/spanner/retry_policy.h"
+#include "google/cloud/spanner/testing/instance_location.h"
+#include "google/cloud/spanner/testing/pick_random_instance.h"
+#include "google/cloud/spanner/testing/random_database_name.h"
+#include "google/cloud/spanner/timestamp.h"
+#include "google/cloud/internal/absl_str_cat_quiet.h"
+#include "google/cloud/internal/getenv.h"
+#include "google/cloud/internal/random.h"
+#include "google/cloud/kms_key_name.h"
+#include "google/cloud/testing_util/integration_test.h"
+#include "google/cloud/testing_util/status_matchers.h"
+#include "absl/time/time.h"
+#include <gmock/gmock.h>
+#include <chrono>
+#include <regex>
+#include <sstream>
+#include <string>
+
+namespace google {
+namespace cloud {
+namespace spanner {
+inline namespace SPANNER_CLIENT_NS {
+namespace {
+
+using ::google::cloud::testing_util::IsOk;
+using ::google::cloud::testing_util::StatusIs;
+using ::testing::HasSubstr;
+using ::testing::IsEmpty;
+using ::testing::Not;
+
+// Constants used to identify the encryption key.
+auto constexpr kKeyRing = "spanner-cmek";
+auto constexpr kKeyName = "spanner-cmek-test-key";
+
+std::string const& ProjectId() {
+  static std::string project_id =
+      google::cloud::internal::GetEnv("GOOGLE_CLOUD_PROJECT").value_or("");
+  return project_id;
+}
+
+bool RunSlowBackupTests() {
+  static bool run_slow_backup_tests =
+      google::cloud::internal::GetEnv(
+          "GOOGLE_CLOUD_CPP_SPANNER_SLOW_INTEGRATION_TESTS")
+          .value_or("")
+          .find("backup") != std::string::npos;
+  return run_slow_backup_tests;
+}
+
+bool Emulator() {
+  static bool emulator =
+      google::cloud::internal::GetEnv("SPANNER_EMULATOR_HOST").has_value();
+  return emulator;
+}
+
+class BackupTest : public ::google::cloud::testing_util::IntegrationTest {
```

Comment:
extra nitty nit: the class name does not match the file name.

Reply:
Done.